evergage/evergage-global-templates
# @W-8307521@ | Redirect

# Description

Adds the Redirect Template (using a [Cumulus dev site](https://cumulusfinserv-dev.herokuapp.com/))

This Global Template redirects the current page to another page, according the provided URL.

## Related Issues/Tickets

JIRA: https://help.evergage.com/browse/EVG-12047
GUS: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008AbFIIA0/view

## How to test
The Template and Campaign can be tested on the homepage of the [Cumulus dev site](https://cumulusfinserv-dev.herokuapp.com/) by forcing the SDK URL for the `evgops` dataset in `training`: https://cdn.evgnet.com/beacon/training/evgops/scripts/evergage.min.js

- **Template name:** [EVG-12047] Global - Redirect (query param opt)
- **Campaign name:** [EVG-12047] Cumulus - Homepage Redirect Test

_Note: it's important to note that this Template should be tightly coupled with [updates to the Flicker Defender gear](https://help.evergage.com/browse/EVG-12209)._

Test Link:
- https://cumulusfinserv-dev.herokuapp.com/
_(Should also try adding query parameters as well, such as `?evergageTestMessages=true&something_else=1234`)_


### Things to check for:
* **FlickerDefender**: Is the page being fully hidden until the following page?
* **Template-specific documentation**: Is there sufficient information for a Template Developer to use the Global Template and customize it for their own use?
* **Campaign stats tracking:** Do impression stats and click stats fire for both the Control and Test experience?


### Additional Notes and Concerns:
- TODO

#### TODOs
- Handlebars comment 
- JS doc comments
- Add logic to protect against a Campaign without rules and that's targeting basically just the domain? (such as https://cumulusfinserv-dev.herokuapp.com/) Maybe using content zones?
